newlib-nano: NEWLIB_INCLUDE_DIR set incorrectly

#### Description
With #9243 merged, I can not build anything for Cortex-M anymore, as apparently the `NEWLIB_INCLUDE_DIR` is not set properly for my environment:
```
cc1: error: /newlib-nano: No such file or directory [-Werror=missing-include-dirs]
cc1: all warnings being treated as errors
```

I might have a more or less custom setup (though not too fancy I would guess):
- toolchain at custom location
- custom folder containing symbolic links to binaries added to global PATH var

Now the definition of the `NEWLIB_INCLUDE_DIR` variable in `makefiles/libc/newlib.mk` does not cover this case, and thus leading to my setup not be able to build anything with the newly enabled warnings set.

Quick solution is for me to export the `NEWLIB_INCLUDE_DIR` variable manually for my environment, but this seems quite hacky to me. So any suggestion on how to fix this in a more nice way?

This is one of the extremely annoying things, forcing me to debug our make system for things that break from one day to the other...

I have a similar problem on  Ubuntu 17.10 now:

```
BOARD=iotlab-m3 IOTLAB_SITE=paris IOTLAB_PHY_NODES=1-11 IOTLAB_DURATION=20 make -C examples/gnrc_networking
make: Entering directory '/home/mlenders/Repositories/RIOT-OS/RIOT/examples/gnrc_networking'
Building application "gnrc_networking" for "iotlab-m3" with MCU "stm32f1".

cc1: error: /etc/alternatives/gcc-arm-none-eabi-include/newlib-nano: No such file or directory [-Werror=missing-include-dirs]
cc1: error: unrecognized command line option '-Wno-implicit-fallthrough' [-Werror]
cc1: all warnings being treated as errors
/home/mlenders/Repositories/RIOT-OS/RIOT/Makefile.base:83: recipe for target '/home/mlenders/Repositories/RIOT-OS/RIOT/examples/gnrc_networking/bin/iotlab-m3/application_gnrc_networking/main.o' failed
make[1]: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/examples/gnrc_networking/bin/iotlab-m3/application_gnrc_networking/main.o] Error 1
/home/mlenders/Repositories/RIOT-OS/RIOT/examples/gnrc_networking/../../Makefile.include:359: recipe for target '/home/mlenders/Repositories/RIOT-OS/RIOT/examples/gnrc_networking/bin/iotlab-m3/application_gnrc_networking.a' failed
make: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/examples/gnrc_networking/bin/iotlab-m3/application_gnrc_networking.a] Error 2
make: Leaving directory '/home/mlenders/Repositories/RIOT-OS/RIOT/examples/gnrc_networking'
```

```
[mlenders@mk RIOT]<3 lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 17.10
Release:	17.10
Codename:	artful
[mlenders@mk RIOT]<3 uname -a
Linux mk 4.13.0-45-generic #50-Ubuntu SMP Wed May 30 08:23:18 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
[mlenders@mk RIOT]<3 ./dist/tools/ci/print_toolchain_versions.sh 
Installed compiler toolchains 
-----------------------------
             native gcc: gcc (Ubuntu 7.2.0-8ubuntu3.2) 7.2.0
      arm-none-eabi-gcc: arm-none-eabi-gcc (15:5.4.1+svn241155-1) 5.4.1 20160919
                avr-gcc: avr-gcc (GCC) 5.4.0
       mips-mti-elf-gcc: missing
             msp430-gcc: msp430-gcc (GCC) 4.6.3 20120301 (mspgcc LTS 20120406 unpatched)
   riscv-none-embed-gcc: missing
                  clang: clang version 4.0.1-6 (tags/RELEASE_401/final)

Installed compiler libs
-----------------------
   arm-none-eabi-newlib: "2.4.0"
    mips-mti-elf-newlib: missing
riscv-none-embed-newlib: missing
               avr-libc: "2.0.0" ("20150208")

Installed development tools
---------------------------
                  cmake: cmake version 3.9.1
               cppcheck: missing
                doxygen: 1.8.13
                 flake8: 3.2.1 (pyflakes: 1.5.0, mccabe: 0.6.1, pycodestyle: 2.3.1) CPython 3.6.3 on Linux
                    git: git version 2.14.1
             coccinelle: missing
```

on macOS I currently get: 
```
BOARD=samr21-xpro make -C examples/gnrc_networking
Warning: no PORT set!
Building application "gnrc_networking" for "samr21-xpro" with MCU "samd21".

cc1: error: -I/RIOT/core/include: No such file or directory [-Werror=missing-include-dirs]
In file included from /RIOT/examples/gnrc_networking/main.c:23:0:
/RIOT/sys/include/shell.h:25:10: fatal error: kernel_defines.h: No such file or directory
 #include "kernel_defines.h"
          ^~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
compilation terminated.
```

when building anything for cortex based boards.

--- a/makefiles/libc/newlib.mk
+++ b/makefiles/libc/newlib.mk
@@ -21,6 +21,13 @@ export LINKFLAGS += -lc
 
 # Search for Newlib include directories
 
+# Try to search for newlib in the standard search path of the compiler for includes
+ifeq (,$(NEWLIB_INCLUDE_DIR))
+  COMPILER_INCLUDE_PATHS := $(shell $(PREFIX)gcc -v -x c -E /dev/null 2>&1 | grep '^\s' | tr -d '\n')
+  NEWLIB_INCLUDE_DIR := $(firstword $(dir $(wildcard $(addsuffix /newlib.h, $(COMPILER_INCLUDE_PATHS)))))
+endif
+
+ifeq (,$(NEWLIB_INCLUDE_DIR))
 # Since Clang is not installed as a separate instance for each crossdev target
 # we need to tell it where to look for platform specific includes (Newlib
 # headers instead of Linux/Glibc headers.)
@@ -35,17 +42,18 @@ export LINKFLAGS += -lc
 # Gentoo crossdev, but we prefer to look at /etc/alternatives first.
 # On OSX, newlib includes are possibly located in
 # /usr/local/opt/arm-none-eabi*/arm-none-eabi/include or /usr/local/opt/gcc-arm/arm-none-eabi/include
-NEWLIB_INCLUDE_PATTERNS ?= \
-  /etc/alternatives/gcc-$(TARGET_ARCH)-include \
-  /usr/$(TARGET_ARCH)/include \
-  /usr/local/opt/$(TARGET_ARCH)*/$(TARGET_ARCH)/include \
-  /usr/local/opt/gcc-*/$(TARGET_ARCH)/include \
-  #
+  NEWLIB_INCLUDE_PATTERNS ?= \
+    /etc/alternatives/gcc-$(TARGET_ARCH)-include \
+    /usr/$(TARGET_ARCH)/include \
+    /usr/local/opt/$(TARGET_ARCH)*/$(TARGET_ARCH)/include \
+    /usr/local/opt/gcc-*/$(TARGET_ARCH)/include \
+    #
 # Use the wildcard Makefile function to search for existing directories matching
 # the patterns above. We use the -isystem gcc/clang argument to add the include
 # directories as system include directories, which means they will not be
 # searched until after all the project specific include directories (-I/path)
-NEWLIB_INCLUDE_DIR ?= $(firstword $(wildcard $(NEWLIB_INCLUDE_PATTERNS)))
+  NEWLIB_INCLUDE_DIR ?= $(firstword $(dir $(wildcard $(addsuffix /newlib.h, $(NEWLIB_INCLUDE_PATTERNS)))))
+endif
 
 # If nothing was found we will try to fall back to searching for a cross-gcc in
 # the current PATH and use a relative path for the includes
@@ -64,7 +72,9 @@ ifeq ($(TOOLCHAIN),llvm)
 endif
 
 ifeq (1,$(USE_NEWLIB_NANO))
-  NEWLIB_NANO_INCLUDE_DIR ?= $(NEWLIB_INCLUDE_DIR)/newlib-nano
+  NEWLIB_NANO_INCLUDE_DIR ?= $(firstword $(wildcard $(NEWLIB_INCLUDE_DIR)newlib-nano \
+                                                    $(NEWLIB_INCLUDE_DIR)newlib/nano \
+                                                    $(NEWLIB_INCLUDE_DIR)nano))
   # newlib-nano overrides newlib.h and its include dir should therefore go before
   # the regular system include dirs.
   INCLUDES := -isystem $(NEWLIB_NANO_INCLUDE_DIR) $(INCLUDES)

